### PR TITLE
Use custom SVGs for commit signing lock icon

### PR DESCRIPTION
**Pull request by CirnoT**
chrome_2020-06-22_16-59-59
chrome_2020-06-22_17-00-34
chrome_2020-06-22_17-01-00
chrome_2020-06-22_17-01-47
chrome_2020-06-22_17-03-37
chrome_2020-06-22_17-03-52
chrome_2020-06-22_17-04-18
chrome_2020-06-22_17-04-34

### CODE REVIEW

**Comment by silverwind** [comment]
Add this to `.editorconfig`:

```
[*.svg]
insert_final_newline = false
```

File: assets/svg/gitea-lock-cog.svg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<svg width="16" height="16" aria-hidden="true" viewBox="0 0 14 16">
```
Comment (silverwind) [inline assets/svg/gitea-lock-cog.svg]:
```suggestion
<svg width="16" height="16" viewBox="0 0 14 16">
```

I assume this is unneccessary? If not, I guess it's best added on the SVG helper functions (go and js), so the SVGs here are only defining shapes.

Comment (silverwind) [reply assets/svg/gitea-lock-cog.svg:1]: Actually, I see we do no further postprocessing so it might be required here for now.

Comment (silverwind) [reply assets/svg/gitea-lock-cog.svg]: Checked the functions, `modules/templates/helper.go` and `web_src/js/utils.js` already add the `aria-hidden` attribute to `<svg>` and I think it makes no sense to have it on `<symbol>`, e.g. right now:

````html
<svg class="svg gitea-lock-cog" width="16" height="16" aria-hidden="true">
    <use xlink:href="#gitea-lock-cog">
      <symbol aria-hidden="true" viewBox="0 0 14 16" id="gitea-lock-cog">
        <path></path>
      </symbol>
    </use>
</svg>

File: assets/svg/gitea-lock.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="16" aria-hidden="true" viewBox="0 0 14 16">
```
Comment (silverwind) [inline assets/svg/gitea-lock.svg]:
```suggestion
<svg width="16" height="16" viewBox="0 0 14 16">
```

File: assets/svg/gitea-unlock.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="16" aria-hidden="true" viewBox="0 0 14 16">
```
Comment (silverwind) [inline assets/svg/gitea-unlock.svg]:
```suggestion
<svg width="16" height="16" viewBox="0 0 14 16">
```

**Comment by silverwind** [comment]
`xmlns` is not necessary for inline SVG content, you can remove it.

**Comment by CirnoT** [comment]
Copying SVG inline from octicons primer website generates it so I've also added it.

**Comment by silverwind** [comment]
Yeah it's needed for standalone display. It seems `svg-sprite-loader` already removes the attribute, so it should be fine.

**Review by silverwind — APPROVED** [review]

File: web_src/less/_repository.less [hunk 4]
```diff
@@ -1479,8 +1479,12 @@
                 margin-right: 0;
             }
 
+            .svg {
```
Comment (lunny) [inline web_src/less/_repository.less:1487]: Why we need this change?

Comment (CirnoT) [reply web_src/less/_repository.less:1487]: To add proper margin on the right of svg inside flex container

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
ping LG-TM

**Pull request merged**